### DISCUSSION TITLE
Let the installGiottoEnvironment() function accept a specified conda path.

### DISCUSSION THREAD

**Discussion by IMJoeyZhu** [giotto-suite/Giotto#517] [Ideas]
Dear Giotto group,

Many thanks for developing this powerful and comprensive tool. 

I noticed that `installGiottoEnvironment()` function does not accept a specified conda path, even though I assigned the `conda_path` to my own conda directory like this:

```r
conda_path <- "/path/to/my/Anaconda3"
```

I checked the script ([python_environment.R][1]) and found out that `installGiottoEnvironment()` function keep redirecting the `conda_path` to a `miniconda_path` generated by `reticulate::miniconda_path()`.

So, I made a few changes to the script. All these codes in the scripts:
```r
conda_path = reticulate::miniconda_path()
```

 were replaced by:
```r
if (!dir.exists(conda_path)) {
  conda_path = reticulate::miniconda_path()
}
```
Then I specify my own conda directory before conducting `installGiottoEnvironment()` function:
```r
conda_path <- "/path/to/my/Anaconda3"
installGiottoEnvironment()
```

Finally, I got these messages, shown below, which illustrate that I successfully install the Giotto Environment to the specified conda path:
```r
# To activate this environment, use
#
#     $ conda activate /path/to/my/Anaconda3/envs/giotto_env
#
# To deactivate an active environment, use
#
#     $ conda deactivate
```

I hope this idea could provide some clue for other people facing the same problem as me.

Best regards,
Jiacheng Zhu

[1]:https://github.com/drieslab/Giotto/blob/94e1366f6aaf24f3ec63df72659eae6afde110ff/R/python_environment.R

**Comment by RubD** [comment]
Hi Jiacheng Zhu,

Thanks for the suggestion and I agree that this would provide additional flexibility. 

I hope you know we already do provide a solution to use your own python path / conda environment, although in this case you would have to install everything yourself. And then you would simply have to provide the python path you want to use to  `createGiottoInstructions` using the `python_path` parameter.

The miniconda was created to make it as easy as possible for everyone not familiar with conda environments and/or python paths. However, as mentioned earlier, providing your own conda environment would be an interesting intermediate solution.

I think @mattobny has already been looking into that.

Best,
Ruben

**Reply by IMJoeyZhu** [reply]
Dear Prof. Ruben,

Thank you for your email and kind reminder! I appreciate your feedback.
I have realized that the Giotto Master Branch is indeed capable of handling the task I was previously inquiring about.
It's great to hear that the Giotto Suite will also be able to do so in the future. Thank you for developing such a powerful tool!
I am really looking forward to your update and utilizing it soon.

Best regards,
Jiacheng Zhu

**Comment by mattobny** [comment]
@IMJoeyZhu Hello! PR #549 on our development branch allows this behavior. This will be merged into Suite shortly.

**Reply by IMJoeyZhu** [reply]
Thanks a lot! Looking forward to utilizing it soon. 😄
